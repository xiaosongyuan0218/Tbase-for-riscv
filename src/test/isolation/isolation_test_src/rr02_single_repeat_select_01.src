#there are three records initialized
#diskey for node1/node2/node3 is 3/5/1 accordingly

setup
{
    create table tbl_xx (i int primary key, j int) distribute by shard(i);
}
setup
{
    insert into tbl_xx values (3, 1), (5, 2), (1, 3);
}

teardown
{
    DROP TABLE tbl_xx;
}

session "s1"
step "s1_begin"     { begin; }
step "s1_set_rr"    { set transaction_isolation = 'repeatable read'; }
step "s1_show_iso"  { show transaction_isolation; }
step "s1_update"    { update tbl_xx set j = 11 where i = diskey1 ; }
step "s1_select"    { select i,j from tbl_xx order by i; }
step "s1_delete"    { delete from tbl_xx  ; }
step "s1_commit"    { commit; }
step "s1_abort"     { abort; }


session "s2"
step "s2_begin"     { begin; }
step "s2_select"    { select i,j from tbl_xx order by i;}
step "s2_update"    { update tbl_xx set j = 22 where i = diskey2 ; }
step "s2_update2"   { update tbl_xx set j = 222 where i = diskey2 ; }
step "s2_upsert"    { insert into tbl_xx values(diskey1,11) on conflict(i) do nothing; }
step "s2_delete"    { delete from tbl_xx where j >= 1 ; }
step "s2_commit"    { commit; }
step "s2_abort"    { abort; }

session "s3"
step "s3_begin"     { begin; }
step "s3_select"    { select i,j from tbl_xx order by i; }
step "s3_update"    { update tbl_xx set j = 33 where i = diskey3 ; }
step "s3_delete"    { delete from tbl_xx where j = 3 ; }
step "s3_commit"    { commit; }
step "s3_abort"    { abort; }

permutation "s1_begin" "s1_set_rr" "s1_show_iso" "s1_select" "s2_update" "s2_select" "s1_select" "s2_update2" "s3_update" "s2_select" "s1_select" "s1_commit" "s1_select"



