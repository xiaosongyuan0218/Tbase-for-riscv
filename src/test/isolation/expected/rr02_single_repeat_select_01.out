Parsed test spec with 3 sessions

starting permutation: s1_begin s1_set_rr s1_show_iso s1_select s2_update s2_select s1_select s2_update2 s3_update s2_select s1_select s1_commit s1_select
step s1_begin: begin;
step s1_set_rr: set transaction_isolation = 'repeatable read';
step s1_show_iso: show transaction_isolation;
transaction_isolation

repeatable read
step s1_select: select i,j from tbl_xx order by i;
i              j              

1              3              
3              1              
5              2              
step s2_update: update tbl_xx set j = 22 where i = 5 ;
step s2_select: select i,j from tbl_xx order by i;
i              j              

1              3              
3              1              
5              22             
step s1_select: select i,j from tbl_xx order by i;
i              j              

1              3              
3              1              
5              2              
step s2_update2: update tbl_xx set j = 222 where i = 5 ;
step s3_update: update tbl_xx set j = 33 where i = 1 ;
step s2_select: select i,j from tbl_xx order by i;
i              j              

1              33             
3              1              
5              222            
step s1_select: select i,j from tbl_xx order by i;
i              j              

1              3              
3              1              
5              2              
step s1_commit: commit;
step s1_select: select i,j from tbl_xx order by i;
i              j              

1              33             
3              1              
5              222            
